src/opal.c

```diff
@@ -2213,10 +2213,22 @@ traverse_ast(node_s *node, FILE *dest_fp)
 void
 add_asm_code (asm_code_e code, int intval, char *label)
 {
-  // Create struct with given values and append to asm_cmd_list[]
-  // TBD
+  /// Create struct with given intval and code
+  asm_cmd_e asm_cmd = { 0 };
+  asm_cmd.intval = intval;
+  asm_cmd.cmd = code;
+
+  /// Add the asm_code label if there is one
+  if (label)
+  {
+      asm_cmd.label = calloc (64, sizeof(char));
+      asm_cmd.label = strdup(label);
+  }
 
-  return;
+  logger(DEBUG, "Added command - cmd: %s, label: %s", asm_cmd.cmd, asm_cmd.label);
```

Comment:
1. `asm_cmd` is an enum so, `%s` modifier will fail, you can access the command name string from the `asm_cmds` array.
2. If `label` is `NULL`, you will hit a `NULL pointer exception`. You can check against it and print "NULL" instead.

A possible solution:
```
logger(DEBUG, "Added command - cmd: %s, label: %s", asm_cmds[asm_cmd.cmd], asm_cmd.label ? asm_cmd.label : "NULL");
```